docs/contracts.rst

```diff
@@ -990,12 +1007,17 @@ virtual method lookup.
 
 .. index:: ! constructor
 
+.. _constructor:
+
 Constructors
 ============
-A constructor is an optional function declared with the ``constructor`` keyword which is executed upon contract creation.
-Constructor functions can be either ``public`` or ``internal``. If there is no constructor, the contract will assume the
-default constructor: ``contructor() public {}``.
 
+A constructor is an optional function declared with the ``constructor`` keyword which is executed upon contract creation, and where you can run contract initialisation code. It always generates code that initialises any state variables you declare to their specified default value if you initialise them inline, or zero if you do not.
+
```

Comment:
Perhaps also note that no code is run to initialize state variables to their default values - this is just the initial data representation of the initial storage state.